student.rb

```diff
@@ -1,10 +1,13 @@
-# student.rb
 require_relative 'person'
 
 class Student < Person
-  def initialize(classroom: 'no classroom', name: 'Unknown', age: nil, parent_permission: true)
-    super(name: name, age: age, parent_permission: parent_permission)
+  attr_accessor :classroom, :name
+
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
+    # @name = name
     @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
```

Comment:
I would suggest creating a setter for classroom attribute as we may need to create a student before assigning a classroom 
Check for a method that serves as the 'belongs-to' side (a student belongs to a classroom). [https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

````ruby
attr_reader :classroom
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
````